src/datasource.ts

```diff
@@ -84,17 +92,22 @@ export default class CogniteDatasource extends DataSourceApi<
     this.templateSrv = getTemplateSrv();
     this.url = url;
     this.project = cogniteProject ?? defaultProject;
-    this.connector = new Connector(
+    const connector = new Connector(
       this.project,
       url,
-      this.backendSrv,
+      defaultFetcher,
       oauthPassThru,
       oauthClientCreds,
       enableTemplates,
       enableEventsAdvancedFiltering,
       enableFlexibleDataModelling,
       enableExtractionPipelines
     );
+    this.initSources(connector);
+  }
+  
+  initSources (connector: Connector) {
+    this.connector = connector;
```

Comment:
Why to move the connector init to a separate method instead of keeping it in the constructor?